### PR TITLE
feat: part 1 support several outputs

### PR DESCRIPTION
**Pull request by cyrinux**
we will still need 
* to deduplicate als
* find way to get monitor serial number and use "make model sn"

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -17,63 +18,109 @@ fn main() {
...
+                                prediction_tx,
+                                user_rx,
+                                als,
+                                true,
+                                cfg.model.clone(),
+                            );
+                            frame_capturer.run(cfg.model, controller)
+                        }
+                    };
+                }),
```
Comment (maximbaz) [inline src/main.rs:116]: ideally this piece of code would look like this, for it to happen you'd also have to change a few places from accepting `String` to accepting `&str` (it's a good practice exactly for this purpose, to be able to send the same string to multiple places without lot of memory copy)

```rust
let model = match capturer_config {
    config::Output::Backlight(cfg) => &cfg.model,
    config::Output::DdcUtil(cfg) => &cfg.model,
};

let controller =
    predictor::Controller::new(prediction_tx, user_rx, als, true, model);
frame_capturer.run(model, controller)
```

File: src/predictor/data.rs [hunk 2]
```diff
@@ -16,36 +20,56 @@ pub struct Entry {
...
+        let filename: String = Self::get_filename(name.to_string());
+        let datadir = dirs::data_dir()
+            .ok_or("Unable to get data dir")?
+            .join("wluma");
+        create_dir_all(datadir.clone())?;
+        Ok(datadir.join(filename))
+    }
+
+    pub fn load(id: &str) -> Self {
+        let path = Self::path(id).unwrap();
```
[the remark this answers is not in the document]
Comment (cyrinux) [reply src/predictor/data.rs:48]: pretty

File: config.toml [hunk 3]
```diff
@@ -17,7 +17,9 @@ thresholds = [ 20, 80, 250, 500, 800 ]
 # [als.none]
 
 # TODO multi-monitor setup is not supported, but wanted!
-[output.backlight.eDP-1]
+[[output.backlight]]
+name = "eDP-1"
```
[the remark this answers is not in the document]
Comment (cyrinux) [reply config.toml:21]: ok!

File: src/main.rs (hunk 1 above)
Comment (cyrinux) [reply src/main.rs:102]: not exactly like this but fixed :)

File: config.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (maximbaz) [reply config.toml]: not done

[the remark this answers is not in the document]
Comment (cyrinux) [reply config.toml]: oh shit  i missread

**Pull request merged**